src/cbc_sdk/platform/alerts.py

```diff
@@ -921,6 +921,65 @@ def _query_implementation(cls, cb, **kwargs):
         return AlertSearchQuery(cls, cb).add_criteria("type", ["INTRUSION_DETECTION_SYSTEM"])
 
 
+class GroupedAlert(PlatformModel):
+    """Represents Grouped alerts."""
+    urlobject = "/api/alerts/v7/orgs/{0}/grouped_alerts"
+    swagger_meta_file = "platform/models/grouped_alert.yaml"
+
+    def __init__(self, cb, model_unique_id, initial_data=None):
+        """
+        Initialize the Grouped Alert object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): ID of the alert represented.
+            initial_data (dict): Initial data used to populate the alert.
+        """
+        super(GroupedAlert, self).__init__(cb, model_unique_id, initial_data)
+        self._most_recent_alert = None
+
+        most_recent_alert = initial_data["most_recent_alert"]
+        if "type" in most_recent_alert:
+            if most_recent_alert["type"] == "CB_ANALYTICS":
+                self._most_recent_alert = CBAnalyticsAlert(cb, most_recent_alert["id"], most_recent_alert)
+            elif most_recent_alert["type"] == "WATCHLIST":
+                self._most_recent_alert = WatchlistAlert(cb, most_recent_alert["id"], most_recent_alert)
+            elif most_recent_alert["type"] == "INTRUSION_DETECTION_SYSTEM":
+                self._most_recent_alert = IntrusionDetectionSystemAlert(cb, most_recent_alert["id"], most_recent_alert)
+            elif most_recent_alert["type"] == "DEVICE_CONTROL":
+                self._most_recent_alert = DeviceControlAlert(cb, most_recent_alert["id"], most_recent_alert)
+            elif most_recent_alert["type"] == "HOST_BASED_FIREWALL":
+                self._most_recent_alert = HostBasedFirewallAlert(cb, most_recent_alert["id"], most_recent_alert)
+            elif most_recent_alert["type"] == "CONTAINER_RUNTIME":
+                self._most_recent_alert = ContainerRuntimeAlert(cb, most_recent_alert["id"], most_recent_alert)
+            else:
+                self._most_recent_alert = Alert(cb, most_recent_alert["id"], most_recent_alert)
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for this alert type.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            GroupAlertSearchQuery: The query object for this alert type.
+        """
+        return GroupedAlertSearchQuery(cls, cb)
+
+    @property
+    def most_recent_alert_(self):
```

Comment:
Something a bit odd - 
when I get the first_grouped_alert it has two attributes -
most_recent_alert which is surprisingly a dict
most_recent_alert_ which is the typed alert object.
Is this a bug? It seems the only field should be most_recent_alert and it should be a typed alert object

Reply:
most_recent_alert is the raw response dict from the _info method on the alert. most_recent_alert_ is the actual class object defined as a property. This is the same way workflow behaves in the alert class, so while it appears redundant this shouldn't be a bug